### ISSUE TITLE
ximea.xiapi read camera image sometimes crush

### ISSUE THREAD

**Issue by CarlDegio** [gelsightinc/gsrobotics#16]
I use showimage and show3d in ros2. Current gsdevice code use:
```
from ximea import xiapi
self.cam = xiapi.Camera(self.dev_id)
self.cam.read()
```
this sometimes occur:
```
global cap_v4l.cpp:1119 tryIoctl VIDEOIO(V4L2:/dev/video2): select() timeout.
ERROR! reading image from camera!
```
it seems that the problem is read image from camera failed. Is there any solution? Such as read image by opencv?

**Comment by debrashure** [comment]
I am in the process of updating the repository to only support the GelSight Mini device.
I plan to check the code in tonight.  I will test the ros2 example. 
Are you using the Mini on Linux?  If so, it shouldn't be using the ximea api.

**Comment by CarlDegio** [comment]
yes, I use mini on linux. I have push a ros2 package version: https://github.com/CarlDegio/gelsight_ros2. two nodes use gsdevice so ximea api is used by default.

**Comment by debrashure** [comment]
I just updated the repository and none of the code uses ximea.  If you update and reinstall gelsight, the issue should go away.

**Comment by CarlDegio** [comment]
Thank you! I will try it later

**Comment by CarlDegio** [comment]
the change seems good. But the crush is still exist. And it just happened randomly and frequently. just like this:
```
ERROR! reading image from camera!
...
AttributeError: 'NoneType' object has no attribute 'shape'
[ WARN:0@23.288] global cap_v4l.cpp:872 requestBuffers VIDEOIO(V4L2:/dev/video2): failed VIDIOC_REQBUFS: errno=19 (No such device)
```
Do you have any suggestions for this, like the frequency to read, or the usb cable.

**Comment by CarlDegio** [comment]
At present, I found that it is likely that the usb plug is not in good status.

**Comment by debrashure** [comment]
If the lights are not staying on all the time, that would indicate a USB problem.

**Comment by debrashure** [comment]
Please send an email to ***@***.******@***.***> with the serial number of the Mini and a short description of the problem.
You can cc me at ***@***.******@***.***> to make sure we take care of the issue.

Thank you,
Debbie


Debra Shure
Principal Software Engineer
GelSight Inc.

978-808-7469
179 Bear Hill Rd. Ste 202
Waltham, MA 02451

**Comment by debrashure** [comment]
Please send an email to mini@gelsight.com with the serial number of the Mini and a short description of the problem.
You can cc me at debra_shure@gelsight.com to make sure we take care of the issue.

Thank you,
Debbie

**Comment by debrashure** [comment]
@CarlDegio were able to resolve your issue ???

**Comment by debrashure** [comment]
Yes, you are correct, the USB connection needs to be reinforced.  We are working on making it better.
If you should run into any issues going forward, please don't hesitate to reach out.
Thanks again for your feedback.
